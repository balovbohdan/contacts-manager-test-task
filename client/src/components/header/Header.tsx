import * as React from 'react';

import {LeftBar} from './LeftBar';
import {RightBar} from './RightBar';

import css from './Header.css';

type Props = {
    toggleRegWindow:()=>void;
    toggleLoginWindow:()=>void;
};

export const Header = (props:Props) =>
    <div className={css.main}>
        <LeftBar/>
        <RightBar
            toggleRegWindow={props.toggleRegWindow}
            toggleLoginWindow={props.toggleLoginWindow}/>
    </div>;