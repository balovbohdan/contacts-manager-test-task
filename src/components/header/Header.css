@import url(../../css/constants.css);

.main {
    width: 100vw;
    height: var(--header-height);

    display: flex;
    align-items: center;
    justify-content: space-evenly;

    padding: 10px;
    background: var(--blue-color);
}