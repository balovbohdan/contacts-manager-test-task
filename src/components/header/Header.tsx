import * as React from 'react';

import {LeftBar} from './LeftBar';
import {RightBar} from './RightBar';

import css from './Header.css';

export const Header = () => (
    <div className={css.main}>
        <LeftBar/>
        <RightBar/>
    </div>
);