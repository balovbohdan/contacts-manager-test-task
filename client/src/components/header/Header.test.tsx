import * as React from 'react';
import {shallow} from 'enzyme';

import {Header} from './Header';

describe('Header', () => {
    it('Should match snapshot', () => {
        const toggleRegWindow = jest.fn();
        const toggleLoginWindow = jest.fn();

        const header = shallow(
            <Header
                toggleRegWindow={toggleRegWindow}
                toggleLoginWindow={toggleLoginWindow}/>
        );

        expect(header).toMatchSnapshot();
    });
});