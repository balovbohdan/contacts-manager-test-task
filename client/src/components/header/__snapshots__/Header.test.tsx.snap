// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `
<div
  className="main"
>
  <Component />
  <Component
    toggleLoginWindow={[MockFunction]}
    toggleRegWindow={[MockFunction]}
  />
</div>
`;
